### PR TITLE
Fix local --icu/--custom-icu configuration

### PR DESCRIPTION
**Pull request by jackhorton**
Unblocks Node-ChakraCore#chakracore-master

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -222,10 +227,15 @@ while [[ $# -gt 0 ]]; do
         ICU_PATH=$1
         # `eval` used to resolve tilde in the path
         eval ICU_PATH="${ICU_PATH:6}"
-        if [[ ! -d $ICU_PATH || ! -d $ICU_PATH/unicode ]]; then
-            # if --custom-icu is given, do not fallback to no-icu
-            echo "!!! couldn't find ICU at $ICU_PATH"
-            exit 1
+        if [[ ! -d "$ICU_PATH" || ! -d "$ICU_PATH/unicode" ]]; then
+            ICU_PATH_LOCAL="$CHAKRACORE_DIR/$ICU_PATH"
```
Comment (obastemur) [inline build.sh]: please define a bash method and use it instead of using multiple copies of the same code.

Comment (jackhorton) [reply build.sh]: Sure

File: build.sh [hunk 2]
```diff
@@ -138,6 +138,22 @@ else
     echo -e "Warning: Installation script couldn't detect host OS..\n" # exit ?
 fi
 
+SET_CUSTOM_ICU() {
+    ICU_PATH=$1
```
Comment (obastemur) [inline build.sh]: Why? `ICU_PATH` is already global.

File: build.sh [hunk 3]
```diff
@@ -138,6 +138,22 @@ else
...
+        ICU_PATH_LOCAL="$CHAKRACORE_DIR/$ICU_PATH"
+        if [[ ! -d "$ICU_PATH_LOCAL" || ! -d "$ICU_PATH_LOCAL/unicode" ]]; then
+            # if a path was explicitly provided, do not fallback to no-icu
+            echo "!!! couldn't find ICU at either $ICU_PATH or $ICU_PATH_LOCAL"
+            exit 1
+        else
+            ICU_PATH="$ICU_PATH_LOCAL"
+        fi
+    fi
+    CMAKE_ICU="-DICU_INCLUDE_PATH_SH=$ICU_PATH"
```
Comment (obastemur) [inline build.sh]: `CMAKE_ICU="-DICU_INCLUDE_PATH_SH=${ICU_PATH}"` instead?

**Comment by obastemur** [comment]
couple of small nits. LGTM

File: build.sh (hunk 3 above)
Comment (jackhorton) [reply build.sh]: any particular reason?

Comment (obastemur) [reply build.sh]: old habit. In such context `${}` is less error-phrone

**Pull request merged**